internal/configs/virtualserver_test.go

```diff
@@ -1906,7 +2014,7 @@ func TestGeneratePoliciesFails(t *testing.T) {
 	vsc := newVirtualServerConfigurator(&ConfigParams{}, false, false, &StaticConfigParams{})
 
 	for _, test := range tests {
-		result := vsc.generatePolicies(owner, ownerNamespace, test.policyRefs, test.policies)
+		result := vsc.generatePolicies(owner, ownerNamespace, vsNamespace, vsName, test.policyRefs, test.policies)
```

Comment:
could you add a case for when you reference two rate limiting policies with conflicting parameters?

To accomplish that, you can test struct with expected warnings:
```

	tests := []struct {
		policyRefs []conf_v1.PolicyReference
		policies   map[string]*conf_v1alpha1.Policy
		expected   policiesCfg
		expectedWarnings Warnings
		msg        string
	}{
```
and then check them like below:
```
if !reflect.DeepEqual(vsc.warnings, test.expectedWarnings) ...
```

Make sure that you have 3 warnings - for dryRun, loglevel and reject code

Note: make sure to recreate `newVirtualServerConfigurator` inside the test loop, because Configurator doesn't clear previous warnings when you call `generatePolicies`.